Introduction to Free Monads/article.md

```diff
@@ -0,0 +1,702 @@
+# Introduction to Free Monads
+
+If you've been around Haskell circles for a bit, you've probably seen the term "free monads".
```

Comment:
This is rather meandering, perhaps just list some of the purposes Free Monads can be used for (as opposed to more common types of monads)?

Reply:
I do go over the uses in the examples and in conclusion, so I don't know if it makes much sense repeating that here -- especially considering the reader likely has no idea of what free monads are yet.

Maybe I'm looking at this wrong, but I find it to be much more natural to first explain the concept in some detail, and then go over how it is useful. That way, the reader/listener can make their own judgement whether it is indeed useful, since they have at least some understanding at that point. Starting off with "here's how you can use that thing which you have no idea about" is rather prevalent, but I personally don't particularly like this approach. In my opinion, it's like starting the book from the back -- you read the payoff the author spent the whole book setting up, but you don't have enough context to appreciate (or sometimes even understand) it.

Reply:
I don't think this explains free monads, right? 

```
If you're relatively new to Haskell, you might feel that monads are hard enough to grasp as is, and now there's these different type of monads to worry about.
Perhaps you've used Haskell for a while but didn't yet get to wrapping your head around free monads.
In any case, the concept is ultimately rather straightforward.
This article aims to introduce free monads and to explain why they are useful.
```

Basically, you want to make people be able to match with their goals as fast as possible. People frequently Google stuff, and this will be the most frequent way that this article will be opened (along with other links from our blog). 

So your goal (and chance in this paragraph) is to say how free monads will matter to the person at the moment when they are Googling: you need to convince them to read the rest of the article. While a person that doesn't need to be convinced will probably just skip to the meat of the article.

Reply:
> Basically, you want to make people be able to match with their goals as fast as possible. People frequently Google random stuff, and this will be the most frequent way that this article will be opened 

➕ 

I think it's a frequent scenario, when you want to implement X and remember (or heard, or assume) that it could be done with Y. Then you google Y and go through relevant links. If you see the article where the first sentences say something like "Y is commonly used for A, B, ... X"  and "we are going to show you what Y is", there is a high chance you will stay on the page. However, when there is only "Y is ..." and nothing about X, the chance is lower.